### PR TITLE
[Optimize] more easily adapt to other views

### PR DESCRIPTION
**Pull request by 838239178**
## Changes

1. use ScrollPositionHelper.kt to reduce computational coupling
2. add max sliding offset to attrs.xml

## Note

It can more easily adapt to other views. I just copy codes from ScrollViewHelper to each ScrollPostionHelper

*I don't recommend adding demo modules to the package*

### CODE REVIEW

**Comment by hannesa2** [comment]
> I don't recommend adding demo modules to the package

What do you mean ?

File: library/src/main/java/com/sothree/slidinguppanel/positionhelper/ScrollPositionHelper.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.sothree.slidinguppanel.positionhelper
+
+import android.view.View
+
+interface ScrollPositionHelper{
```
Comment (hannesa2) [inline library/src/main/java/com/sothree/slidinguppanel/positionhelper/ScrollPositionHelper.kt:5]: It's not important, but I guess this file needs a format. `...lper {`

Comment (838239178) [reply library/src/main/java/com/sothree/slidinguppanel/positionhelper/ScrollPositionHelper.kt:5]: > What do you mean ?

may delete this paragraph in `setting.gradle` when deploying to maven or others
```gradle
include ':demo'
```

Comment (hannesa2) [reply library/src/main/java/com/sothree/slidinguppanel/positionhelper/ScrollPositionHelper.kt:5]: I already did the format. My indicator was the missing blank between `ScrollPositionHelper` and `{`

Comment (hannesa2) [reply library/src/main/java/com/sothree/slidinguppanel/positionhelper/ScrollPositionHelper.kt:5]: > may delete this paragraph in `setting.gradle` when deploying to maven or others
> 
> ```groovy-gradle
> include ':demo'
> ```

No action required, only `library` will be deployed  https://github.com/hannesa2/AndroidSlidingUpPanel/blob/master/library/build.gradle#L4

Comment (838239178) [reply library/src/main/java/com/sothree/slidinguppanel/positionhelper/ScrollPositionHelper.kt:5]: > https://github.com/hannesa2/AndroidSlidingUpPanel/blob/master/library/build.gradle#L4

sry, this is my problem

**Pull request merged**